pkg/runtime/k0s/runtime.go

```diff
@@ -0,0 +1,197 @@
+// Copyright © 2022 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k0s
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync"
+
+	"github.com/sealerio/sealer/common"
+	"github.com/sealerio/sealer/pkg/registry"
+	"github.com/sealerio/sealer/pkg/runtime"
+	"github.com/sealerio/sealer/pkg/runtime/k0s/k0sctl"
+	v2 "github.com/sealerio/sealer/types/api/v2"
+	utilsnet "github.com/sealerio/sealer/utils/net"
+	"github.com/sealerio/sealer/utils/platform"
+	"github.com/sealerio/sealer/utils/ssh"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/sirupsen/logrus"
+)
+
+type Config struct {
+	Vlog      int
+	RegConfig *registry.Config
+	// 	clusterFileK0sConfig is k0sctl cluster config file
+	K0sConfig *k0sctl.K0sConfig
+}
+
+// Runtime struct is the runtime interface for k0s
+type Runtime struct {
+	*sync.Mutex
+	// cluster is sealer clusterFile
+	cluster *v2.Cluster
+	*Config
+}
+
+func (k *Runtime) Init() error {
+	return k.init()
+}
+
+func (k *Runtime) Upgrade() error {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (k *Runtime) Reset() error {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (k *Runtime) JoinMasters(newMastersIPList []net.IP) error {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (k *Runtime) JoinNodes(newNodesIPList []net.IP) error {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (k *Runtime) DeleteMasters(mastersIPList []net.IP) error {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (k *Runtime) DeleteNodes(nodesIPList []net.IP) error {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (k *Runtime) GetClusterMetadata() (*runtime.Metadata, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (k *Runtime) UpdateCert(certs []string) error {
+	//TODO implement me
+	panic("implement me")
+}
+
+// NewK0sRuntime arg "clusterConfig" is the k0s config file under etc/${ant_name.yaml}, runtime need read k0s config from it
+// Mount image is required before new Runtime.
+func NewK0sRuntime(cluster *v2.Cluster) (runtime.Interface, error) {
+	return newK0sRuntime(cluster)
+}
+
+func newK0sRuntime(cluster *v2.Cluster) (runtime.Interface, error) {
+	k := &Runtime{
+		cluster: cluster,
+		Config:  &Config{},
+	}
+
+	k.Config.RegConfig = registry.GetConfig(k.getImageMountDir(), k.cluster.GetMaster0IP())
+
+	if err := k.checkList(); err != nil {
+		return nil, err
+	}
+
+	if logrus.GetLevel() == logrus.DebugLevel {
```

Comment:
How about create a function for level transform?

Reply:
This would be fine, more readable.